### PR TITLE
History view and paging

### PR DESCRIPTION
**Pull request by nikk1e**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -174,6 +174,7 @@ app.get('/login', function(req, res, next) {
 //TODO: authenticate/register with Google/Twitter/Facebook
 
 app.use('/:area/:project', project);
+app.use('/:area/:project/history/:page*', project);
```
Comment (nikk1e) [inline app.js]: Added pages to the routing

File: routes/project.js [hunk 2]
```diff
@@ -110,7 +111,7 @@ function tree(req, res, next) {
 				url: baseUrl + branch + urlPath + '/' + (isDir ? name + '/' : name.replace(/.cube$/,''))
 			});
 		}
-		res.render('tree', {title: req.params.area + '/' + project + (urlPath ? '/' + urlPath : ''), tree: treeM, commit: commit});
+		res.render('tree', {title: req.params.area + '/' + project + (urlPath ? '/' + urlPath : ''), tree: treeM, commit: commit, historyUrl: historyUrl});
```
Comment (nikk1e) [inline routes/project.js]: Pass the history link to the area

File: routes/project.js [hunk 3]
```diff
@@ -250,18 +251,37 @@ router.get('/:branch/*.xslx', function(req, res, next) {
 	res.send('I am xslx for ' + req.params[0]);
 });
 
-router.get('/:branch/history', function(req, res, next) {
+router.get('/:branch/history/:page?', function(req, res, next) {
```
Comment (nikk1e) [inline routes/project.js]: Paging

File: views/history.jade [hunk 4]
```diff
@@ -0,0 +1,32 @@
+extends layout
```
Comment (nikk1e) [inline views/history.jade]: I couldnt see a history view when i pulled your code down so created one

File: routes/project.js [hunk 5]
```diff
@@ -250,18 +251,37 @@ router.get('/:branch/*.xslx', function(req, res, next) {
...
+		function slicer(items){
+			var offset = (itemsPerPage*(page-1));
+			if(items.length > offset){
+				return items.slice(offset);
+			}
+			return items;
+		}
+
+		res.render('history', {	commits: slicer(commits),
+								title: title,
```
Comment (mavxg) [inline routes/project.js]: This is not how you should do paging for a github repository. As it is currently written, you would need the history function to scan the full history of the repository every time. A better option would be to just render the url of the next button as a link the history of the last commit hash from commits (and drop the last commit from the display - unless you like the one commit overlap).

File: app.js [hunk 6]
```diff
@@ -174,6 +174,7 @@ app.get('/login', function(req, res, next) {
 //TODO: authenticate/register with Google/Twitter/Facebook
 
 app.use('/:area/:project', project);
+app.use('/:area/:project/history/:limit*', project);
```
Comment (nikk1e) [inline app.js]: optional limit param

File: routes/project.js [hunk 7]
```diff
@@ -250,18 +251,28 @@ router.get('/:branch/*.xslx', function(req, res, next) {
...
 	branch = branch.replace(/^~/, ''); //remove leading ~ when we match with branch with command
 	
 	var area = req.params.area;
-	var project = req.params.project;
+	var project = req.params.project;	
+	var limit = parseInt(req.params.limit || 20);
 
 	var title = area + '/' + project + (branchR ? '/' + branchR : '')
-	history(req, branch, function(err, commits) {
+	history(req, branch, limit+1, function(err, commits) {
```
Comment (nikk1e) [inline routes/project.js]: Ask for one more item than needed.  Pop it and use the URL as the pointer for the next history page

File: public/css/clay.css [hunk 8]
```diff
@@ -314,4 +314,4 @@ td.error {
...
...position:-384px -120px}.icon-folder-open{width:16px;background-position:-408px -120px}.icon-resize-vertical{background-position:-432px -119px}.icon-resize-horizontal{background-position:-456px -118px}.icon-hdd{background-position:0 -144px}.icon-bullhorn{background-position:-24px -144px}.icon-bell{background-position:-48px -144px}.icon-certificate{background-position:-72px -144px}.icon-thumbs-up{background-position:-96px -144px}.icon-thumbs-down{background-position:-120px -144px}.icon-hand-right{background-position:-144px -144px}.icon-hand-left{background-position:-168px -144px}.icon-hand-up{background-position:-192px -144px}.icon-hand-down{background-position:-216px -144px}.icon-circle-arrow-right{background-position:-240px -144px}.icon-circle-arrow-left{background-position:-264px -144px}.icon-circle-arrow-up{background-position:-288px -144px}.icon-circle-arrow-down{background-position:-312px -144px}.icon-globe{background-position:-336px -144px}.icon-wrench{background-position:-360px -144px}.icon-tasks{background-position:-384px -144px}.icon-filter{background-position:-408px -144px}.icon-briefcase{background-position:-432px -144px}.icon-fullscreen{background-position:-456px -144px}
```
Comment (nikk1e) [inline public/css/clay.css]: Typo "d,isplay"

**Pull request merged**